tests/integration/security/mtls_cert_securenaming/mtls_cert_securenaming.go

```diff
@@ -125,9 +106,124 @@ spec:
 			}
 			checker.CheckOrFail(ctx)
 			t.Log("mTLS check passed")
+
+			t.Log("Secure naming testing #1: connection fails when DR doesn't match SA.")
+			defaultIdentityDR := fmt.Sprintf(`apiVersion: "networking.istio.io/v1alpha3"
+kind: "DestinationRule"
+metadata:
+  name: "service-b-dr"
+spec:
+  host: "b.%s.svc.cluster.local"
+  trafficPolicy:
+    tls:
+      mode: ISTIO_MUTUAL
+      subjectAltNames:
+      - "spiffe://cluster.local/ns/%s/sa/default"
+`, testNamespace.Name(), testNamespace.Name())
+			g.ApplyConfigOrFail(t, testNamespace, defaultIdentityDR)
+			time.Sleep(time.Second * 3) // Wait for the DR to take effect
+			checker = connection.Checker{
+				From:          a,
+				Options:       callOptions,
+				ExpectSuccess: false,
+			}
+			checker.CheckOrFail(ctx)
+
+			t.Log("Secure naming testing #2: connection succeeds when DR matches SA.")
```

Comment:
These can be subtests with ctx.NewSubtest (or similar) instead of relying on logging to separate them

Reply:
And can also be table driven

Reply:
Good point